packages/web3-utils/src/converters.ts

```diff
@@ -0,0 +1,375 @@
+/**
+ * @alias `utf8ToHex`
+ */
+export const stringToHex = utf8ToHex;
+
+/**
+ * Should be called to get utf8 from it's hex representation
+ */
+export const hexToUtf8 = (str: HexString): string => bytesToBuffer(str).toString('utf8');
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const toUtf8 = hexToUtf8;
+
+/**
+ * @alias `hexToUtf8`
+ */
+export const hexToString = hexToUtf8;
+
+/**
+ * Should be called to get hex representation (prefixed by 0x) of ascii string
+ */
+export const asciiToHex = (str: string): HexString => {
+	validateStringInput(str);
+
+	return `0x${Buffer.from(str, 'ascii').toString('hex')}`;
+};
+
+/**
+ * @alias `asciiToHex`
+ */
+export const fromAscii = asciiToHex;
+
+/**
+ * Should be called to get ascii from it's hex representation
+ */
+export const hexToAscii = (str: HexString): string => bytesToBuffer(str).toString('ascii');
+
+/**
+ * @alias `hexToAscii`
+ */
+export const toAscii = hexToAscii;
+
+/**
+ * Auto converts any given value into it's hex representation.
+ */
+export const toHex = (
+	value: Numbers | Bytes | Address | boolean,
+	returnType?: boolean,
+): HexString | ValueTypes => {
+	if (typeof value === 'string' && isAddress(value)) {
+		return returnType ? 'address' : `0x${value.toLowerCase().replace(/^0x/i, '')}`;
+	}
+
+	if (typeof value === 'boolean') {
+		// eslint-disable-next-line no-nested-ternary
+		return returnType ? 'bool' : value ? '0x01' : '0x00';
+	}
+
+	if (typeof value === 'number') {
+		// eslint-disable-next-line no-nested-ternary
+		return returnType ? (value < 0 ? 'int256' : 'uint256') : numberToHex(value);
+	}
+
+	if (typeof value === 'bigint') {
+		return returnType ? 'bigint' : numberToHex(value);
+	}
+
+	if (typeof value === 'string') {
+		if (value.startsWith('-0x') || value.startsWith('-0X')) {
+			return returnType ? 'int256' : numberToHex(value);
+		}
+
+		if (isHexStrict(value)) {
+			return returnType ? 'bytes' : value;
+		}
+
+		if (!Number.isFinite(value)) {
+			return returnType ? 'string' : utf8ToHex(value);
+		}
+	}
+
+	throw new Error(errorCannotParseHex(value));
+};
+
+/**
+ * Auto converts any given value into it's hex representation,
+ * then converts hex to number.
+ */
+export const toNumber = (value: Numbers): number | bigint => {
+	if (typeof value === 'number') {
+		return value;
+	}
+
+	if (typeof value === 'bigint') {
+		return value >= Number.MIN_SAFE_INTEGER || value <= Number.MAX_SAFE_INTEGER
+			? Number(value)
+			: value;
+	}
+
+	return hexToNumber(numberToHex(value));
+};
+
+/**
+ * Takes a number of wei and converts it to any other ether unit.
+ */
+export const fromWei = (number: Numbers, unit: EtherUnits): string => {
+	const denomination = ethUnitMap[unit];
+
+	if (!denomination) {
+		throw new Error(errorInvalidUnit(unit));
+	}
+
+	// count number of zeros in denomination
+	// 1000000 -> 6
+	const numberOfZerosInDenomination = denomination.toString().length - 1;
+
+	if (numberOfZerosInDenomination <= 0) {
+		return number.toString();
+	}
```

Comment:
I don't understand what's happening here, could you clarify with a comment?

Reply:
Oh I think I understand, this handles if the provided `unit` is `wei` or `noether`?

Reply:
Yes you are right.